xdem/spstats.py

```diff
@@ -9,19 +9,92 @@
 import random
 import warnings
 from functools import partial
-from typing import Callable, Union
+from typing import Callable, Union, Optional
 
 import matplotlib.pyplot as plt
 import numpy as np
 import pandas as pd
 from scipy import integrate
 from scipy.optimize import curve_fit
+from skimage.draw import disk
+from scipy.interpolate import RegularGridInterpolator, LinearNDInterpolator, griddata
+from xdem.spatial_tools import nmad
 
 with warnings.catch_warnings():
     warnings.filterwarnings("ignore", category=DeprecationWarning)
     import skgstat as skg
     from skgstat import models
 
+def robust_interp_nd_binning(df: pd.DataFrame, var_names: list[str], statistic : Union[str, Callable, None] = nmad, min_count: Optional[int] = 100) -> Callable:
+    """
+    Estimate an interpolated function for an N-dimensional binning.
+    Relies on scipy.ndimage.map_coordinates: see for more information on input arguments.
+
+    :param df: dataframe of binned values according to explanatory variables
+    :param var_names: list of variable names to use
+    :param statistic: statistic to use
+    :param min_count: filter binned values with a sample count of less than this minimum count
+    :return:
```

Comment:
Here it would be nice with a super-minimal example to show the basic functionality.

```python
"""
:examples:
    >>> df = pd.DataFrame({"var1": [1, 2, 3], "var2": [4, 5, 6], "target": [5, 7, 9]}, min_count=None)
    >>> fun = robust_interp_nd_binning(df, var_names=["var1", "var2"], statistic="target")
    >>> fun((1, 4))
        array(5.)
```
This is just what I assume would work. I haven't tried it.

You can try it with:
```bash
python -m doctest xdem/spstats.py
```

I think such an example (or one that actually works!) would be nice to show the absolute basics of the function!

Reply:
Added in 26d3b9e